runtime/doc/dev_style.txt

```diff
@@ -0,0 +1,47 @@
+*dev_style.txt*          Nvim
+
+
+                            NVIM REFERENCE MANUAL
+
+
+Neovim style guide					*style* *dev-style*
+
+This reference describes the neovim style guide.
+
+                                      Type |gO| to see the table of contents.
+
+==============================================================================
+Background~						*style-background*
+
+One way in which we keep the code base manageable is by enforcing
+consistency.  It is very important that any programmer be able to look at
+another's code and quickly understand it.
+
+Maintaining a uniform style and following conventions means that we can more
+easily use "pattern-matching" to infer what various symbols are and what
+invariants are true about them.  Creating common, required idioms and patterns
+makes code much easier to understand.
+
+In some cases there might be good arguments for changing certain style rules,
+but we nonetheless keep things as they are in order to preserve consistency.
+
+
+The #define Guard~					*style-define-guard*
```

Comment:
probably don't need help tags for each style case. because they might change or be deleted (if/when we automate them), and we don't really want direct references to them.

Reply:
Makes sense. I'll take it easy with the help tags.